pkg/api/imagebasedgroupupgrades/v1alpha1/types.go

```diff
@@ -0,0 +1,104 @@
+type ImageBasedUpgradeAction struct {
+	// +kubebuilder:validation:Enum=Abort;Prep;Upgrade;Rollback;Finalize
+	Action string `json:"action"`
+}
+
+func (i ImageBasedUpgradeAction) String() string {
+	return i.Action
+}
+
+var (
+	// Prep defines the preparing stage for image based upgrade
+	Prep = ImageBasedUpgradeAction{"Prep"}
+	// Upgrade defines the upgrading stage for image based upgrade
+	Upgrade = ImageBasedUpgradeAction{"Upgrade"}
+	// Rollback defines the rollback stage for image based upgrade
+	Rollback = ImageBasedUpgradeAction{"Rollback"}
+	// Finalize defines the finalizing stage for image based upgrade
+	Finalize = ImageBasedUpgradeAction{"Finalize"}
+	// Abort defines the aborting stage for image based upgrade
+	Abort = ImageBasedUpgradeAction{"Abort"}
+	// Unknown defines an unknown ImageBasedUpgrade action
+	Unknown = ImageBasedUpgradeAction{""}
+)
+
+// RolloutStrategy defines how to rollout ibu
+type RolloutStrategy struct {
+	//kubebuilder:validation:Minimum=1
+	MaxConcurrency int `json:"maxConcurrency"`
+	//+kubebuilder:default=240
+	Timeout int `json:"timeout,omitempty"`
+}
+
+// ImageBasedGroupUpgradeSpec defines the desired state of ImageBasedGroupUpgrade
+type ImageBasedGroupUpgradeSpec struct {
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="IBU Spec"
+	//+kubebuilder:validation:Required
+	IBUSpec lcav1alpha1.ImageBasedUpgradeSpec `json:"ibuSpec"`
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Clusters",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	Clusters []string `json:"clusters,omitempty"`
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Cluster Label Selectors",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	ClusterLabelSelectors []metav1.LabelSelector `json:"clusterLabelSelectors,omitempty"`
+	//kubebuilder:validation:Minimum=1
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Actions"
+	// kubebuilder:validation:XValidation:message="Invalid list of actions", rule="self.map(x,x.action)==['Prep'] || self.map(x,x.action)==['Prep','Upgrade'] || self.map(x,x.action)==['Prep','Upgrade','Finalize'] || self.map(x,x.action)==['Rollback'] || self.map(x,x.action)==['Rollback', 'Finalize'] || self.map(x,x.action)==['Upgrade'] || self.map(x,x.action)==['Upgrade','Finalize'] || self.map(x,x.action)==['Finalize'] || self.map(x,x.action)==['Abort'] || self.map(x,x.action)==['Prep', 'Abort'] || self.map(x,x.action)==['Prep', 'Upgrade', 'Rollback'] || self.map(x,x.action)==['Prep', 'Upgrade', 'Rollback', 'Finalize'] || self.map(x,x.action)==['Upgrade','Rollback'] || self.map(x,x.action)==['Upgrade','Rollback','Finalize']"
```

Comment:
I just think of the case where stages are done in multiple IBGUs. For example, IBGU1 has only prep stage. Instead of appending to the existing IBGU1, another IBGU2 is created with upgrade stage.  Do we block/support this case? I suppose ideally we would like a single IBGU to manage the full lifecycle of IBU for the same group of clusters?